homeworks/nagynorbert/4_exceptions_logging/to_do.py

```diff
@@ -0,0 +1,117 @@
+# Homework - Nagy Norbert
+import setup_logger
+import sys
+
+taskfile="homeworks/nagynorbert/4_exceptions_logging/tasks.txt"
+logger = setup_logger.setup_logger("root_logger")
+
+def add_task(task):
+    """Add one task to taskfile."""
+    logger.info(f"Following task will be added to taskfile: {task}")
+    try:    
+        with open(taskfile,"a") as file:
+            file.write(task+"\n")
+    except FileNotFoundError as e:
+        logger.info(f"File not found: {e}")
+    except Exception as e:
+        logger.info(f"Unexpected error happened during adding a task to file: {e}")
+    else:
+        logger.info(f"Add {task} task to {taskfile} was success.")
+
+def view_tasks():
+    try:
+        with open(taskfile,"r") as file:
+            lines = file.readlines()
+    except FileNotFoundError as e:
+        logger.info(f"File not found: {e}")
+    except Exception as e:
+        logger.info(f"Unexpected error happened during viewing taskfile: {e}")
+    else:
+        if len(lines) == 0:
+            logger.info("Task file is empty")
+        else:
+            logger.info(f"{taskfile} content is:")
+            for line in lines:
+                print(line.strip())
+
+def remove_task(task):
+    """Remove one task from taskfile."""
+    logger.info(f"Following task will be removed: {task}")
+    try:
+        with open(taskfile,"r") as file:
+            lines = file.readlines()
+        logger.info(f"Tasks before remove operation:")
+        view_tasks()
+    except Exception as e:
+        logger.info(f"Exception happened during {taskfile} reading: {e}")
+    else:
+            with open(taskfile,"w") as file:
```

Comment:
ez az egész context manager egy tabbal beljebb van mint kéne (lefut így is, de maradj konzisztens kérlek az indentation-nel). Ezt egy kicsit lehetne úgy egyszerűsíteni, hogy ha már amúgyis beolvastad a sorokat egy lines nevű listába, akkor szimplán a listából törölhetnéd a megadott taskot, és visszaírhatnád a file-ba.

Reply:
Köszi az észrevételeket!
Jogos, erre figyelni kell, mert összetett kód esetén totál káosz alakul ki. Javítottam az intendation-t.
Egyszerűsítettem a lista kezeléssel, pár sorral kevesebb lett a remove_task() function.